### PR TITLE
LFS support to be stored on minio

### LINKED ISSUE #5530: move lfs to S3 (minio)
It would be nice to have minio backend for lfs

### PR DESCRIPTION
**Pull request by lunny**
As a next step of #11387, Fix #5530

### CODE REVIEW

File: modules/storage/storage.go [hunk 1]
```diff
@@ -71,3 +97,31 @@ func Init() error {
...
+			minio.AccessKeyID,
+			minio.SecretAccessKey,
+			minio.Bucket,
+			minio.Location,
+			minio.BasePath,
+			minio.UseSSL,
+		)
+	default:
+		return fmt.Errorf("Unsupported LFS store type: %s", setting.LFS.StoreType)
+	}
```
Comment (zeripath) [inline modules/storage/storage.go:120]: Is there really no way to make these self-register?

Comment (lunny) [reply modules/storage/storage.go:120]: I don't know. But the current codes look clear and simple.

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
hmm... I'm wondering if we'd be better off calving off the storage options into:

```ini
[storage] ; this is the default storage option
TYPE=local
PATH=/data ; specific storage will inherit from this as a subpath

[storage.attachments]
TYPE=local; minio or the like
; PATH would default to [storage] PATH/attachments
... ; specific storage content config

[storage.lfs]
TYPE=local; minio or the like
... ; specific storage content config

...
```
Then we can move the specific configuration to the storage module and do something like queues configuration stuff.

**Comment by lunny** [comment]
@zeripath Yes, that's my next PR. But I will only add `[storage]` and keep other configuration on `[attchements]`, `[lfs]` and etc.

**Review by mrsdizzie — APPROVED** [review]

**Pull request merged**